### PR TITLE
Shaul/per 7653 better error messages in the sidecar

### PR DESCRIPTION
**Pull request by shaulk**

### CODE REVIEW

**Review by omer9564 — APPROVED** [review]

File: horizon/opal_relay_api.py [hunk 1]
```diff
@@ -122,19 +138,39 @@ async def send_ping(self):
...
-                raise RuntimeError(
-                    f"Server responded to token request with a bad status {response.status}: {text}"
+                try:
+                    text = await response.text()
+                except:
+                    text = None
+                raise RelayAPIException(
+                    "relay-api",
+                    response.status,
+                    f"Server responded to token request with a bad status {response.status}: {text}",
```
Comment (asafc) [inline horizon/opal_relay_api.py:148]: @shaulk i don't think this error message is correct.
this looks like a copy-paste issue.

also it will create a super verbose message instead of simply showing the status code we got.

File: horizon/opal_relay_api.py [hunk 2]
```diff
@@ -122,19 +138,39 @@ async def send_ping(self):
...
                 )
         logger.debug("Sent ping.")
 
     async def _run(self):
+        backoff = 0
```
Comment (asafc) [inline horizon/opal_relay_api.py:153]: why did you add a backoff functionality?
- this will make resending failed updates much much slower. we don't want that.

File: horizon/opal_relay_api.py [hunk 3]
```diff
@@ -122,19 +138,39 @@ async def send_ping(self):
...
         while True:
             try:
                 await self.send_ping()
-            except Exception:
-                logger.opt(exception=True).warning("Unable to send ping:")
-            await asyncio.sleep(sidecar_config.PING_INTERVAL)
+                backoff = 0
+            except RelayAPIException as e:
+                logger.warning(
+                    "Could not report uptime status to server: {}. This does not affect the PDP's operational state or data updates.",
```
Comment (asafc) [inline horizon/opal_relay_api.py:160]: full error should be (this is an example):
Could not report uptime status to server: got HTTP status code 503. This does not affect the PDP's operational state or data updates.

if you keep using the original e.message you can have a much longer message, that is not necessary.

**Review by asafc — CHANGES_REQUESTED** [review]

**Review by asafc — APPROVED** [review]

File: horizon/opal_relay_api.py (hunk 1 above)
Comment (shaulk) [reply horizon/opal_relay_api.py]: Fixed.
